core/state_transition.go

```diff
@@ -748,6 +765,24 @@ func (st *stateTransition) refundGas(refundQuotient uint64) uint64 {
 	return refund
 }
 
+func (st *stateTransition) refundOperatorCost() {
+	if optimismConfig := st.evm.ChainConfig().Optimism; optimismConfig != nil && !st.msg.IsDepositTx && st.evm.ChainConfig().IsOptimismIsthmus(st.evm.Context.Time) {
+		// Return ETH to transaction sender for operator cost overcharge.
+		operatorCostGasLimit := st.evm.Context.OperatorCostFunc(st.msg.GasLimit, st.evm.Context.Time)
+		operatorCostGasUsed := st.evm.Context.OperatorCostFunc(st.gasUsed(), st.evm.Context.Time)
+
+		if operatorCostGasUsed.Cmp(operatorCostGasLimit) > 0 { // Sanity check.
+			panic(fmt.Sprintf("operator cost gas used (%d) > operator cost gas limit (%d)", operatorCostGasUsed, operatorCostGasLimit))
+		}
+
+		diff := new(uint256.Int).Sub(operatorCostGasLimit, operatorCostGasUsed)
+
+		if diff.Uint64() > 0 {
+			st.state.AddBalance(st.msg.From, diff, tracing.BalanceIncreaseGasReturn)
+		}
```

Comment:
we need to make sure this is aligned with other impls tracing, that this is only called if the diff is strictly >0

Reply:
@Inphi what do you think about the current impl, in view of [your comment](https://github.com/ethereum-optimism/op-geth/pull/388#discussion_r1953211673) above?

Reply:
This is OK as `refundOperatorCost` is only called when isthmus is activated. Though I'd prefer that the function name were a bit more explicit it's isthmus only so it's less likely to be misused in the future.

Reply:
renamed to `refundIsthmusOperatorCost`

Reply:
After speaking to @clabby on this as well, let's always call `AddBalance`, even if `diff` is 0. This is more uniform and in line with the Rust impl.

Reply:
Done